code/controllers/subsystem/achievements.dm

```diff
@@ -10,6 +10,14 @@ SUBSYSTEM_DEF(achievements)
 	var/list/datum/award/score/scores = list()
 	///List of all awards
 	var/list/datum/award/awards = list()
+	/// List of all currently queued up NON-ADDITIVE achievement query arguments (for roundend purposes).
+	var/list/achievement_query_arguments = list()
```

Comment:
It might be "cleaner", but at the same time, I would have to then iterate over every single achievement of every single player in the game, which I feel is a lot more wasteful than this implementation.

Reply:
Alright, i just don't like exposing query parts more than necessary. Could just keep achievement list instead ?

Reply:
That's a good idea, yeah. I'll do that instead.